x/superfluid/keeper/stake.go

```diff
@@ -662,3 +664,49 @@ func (k Keeper) IterateDelegations(ctx sdk.Context, delegator sdk.AccAddress, fn
 		fn(index+int64(i), delegation)
 	}
 }
+
+func (k Keeper) LockExistingFullRangePositionAndSFStake(ctx sdk.Context, positionId uint64, sender sdk.AccAddress, valAddr string) (concentratedLockID uint64, err error) {
```

Comment:
just curious: what's the full context behind this message? Is it so that we can stake CL positions and get staking rewards?

Reply:
if so, then is there a reason why it is only for full range position? and should we remove the locks when user undelegates?

Reply:
The only use case for locking in the context of CL is to SFS. The context behind this message is that we currently only allow direct create and SFS CL position messages, which means if someone had a full range position and wanted to SFS it, they would need to remove the position and create a new one, which is not a great UX. We also can only SFS a full range position as of now, so locking any other position (non full range) would not be a valid action.

Reply:
sounds good, can we please add this in the comment?